src/App.tsx

```diff
@@ -1,61 +1,123 @@
+import React, { useState } from 'react';
+
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+const getUserById = (userId: number) =>
+  usersFromServer.find(user => user.id === userId) as User;
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const todosWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [userID, setUserID] = useState(0);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [errorTitle, setErrorTitle] = useState('');
+  const [errorID, setErrorID] = useState('');
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const getMaxId = () => Math.max(...todos.map(todo => todo.id));
```

Comment:
Better move this function outside the component

Reply:
> Better move this function outside the component

But if I take it out of the component, I won't have access to the state, right?